flocker/node/_deploy.py

```diff
@@ -934,6 +935,13 @@ def _restart_for_application_change(
         # we don't want to do anything:
         comparable_state = comparable_state.transform(["running"], True)
 
+        # Restart policies were briefly supported but they interact poorly with
```

Comment:
And fixes the case where the application is running with "always" restart policy but the configuration is changed to "never" simultaneously with the Flocker upgrade.